File: python/x.py
from ashtlog import PlainLogBackend


def main():
    a = PlainLogBackend()
    b = a.child("b")
    b.put("1")
    c = b.child_shared("c")
    c.put("2")
    d = b.child_shared("d")
    d.put("3")
    e = a.child("e")
    e.put("4")
    f = e.child_shared("f")
    f.put("5")
    a.put("6")

    g = a.child("g")
    h = g.child_shared("h")
    i = h.child_shared("i")
    j = i.child("j")
    k = j.child("k")
    k.put("7")


if __name__ == "__main__":
    main()

File: nim/x.nim
#!/usr/bin/env long-shebang
#!nim c -r

func escape(x: string): string =
    for c in x:
        if c == '/' or c == '>' or c == '\\':
            result.add('\\')
        result.add(c)

type LogBackend = object of RootObj

type PlainLogBackend {.final.} = object of LogBackend
    parent: ref PlainLogBackend
    name: string

proc put(self: ref PlainLogBackend, entry: string) =
    var segments: seq[string] = @[]
    var here = self
    while here != nil:
        if here.name == "":
            segments.add(">")
        else:
            segments.add(escape(here.name))
    echo entry

# echo escape(r"/h\i>>")
# let p = new PlainLogBackend(parent: nil, name: "")
# p.put("hihihi")
let x = PlainLogBackend.ee
